core/src/commonMain/kotlin/com/xebia/functional/xef/auto/AIRuntime.kt

```diff
@@ -1,9 +1,11 @@
 package com.xebia.functional.xef.auto
 
-import com.xebia.functional.xef.embeddings.OpenAIEmbeddings
-import com.xebia.functional.xef.env.OpenAIConfig
-import com.xebia.functional.xef.llm.openai.KtorOpenAIClient
-import com.xebia.functional.xef.llm.openai.LLMModel
+import com.aallam.openai.client.OpenAI
+import com.aallam.openai.client.OpenAIConfig
```

Comment:
I thought the goal of the new `openai` module, and this effort was to abstract away OpenAI from `core`.
So we shouldn't be relying directly on `openai` from `core`, but instead only on `AIClient`.

So this `openAI()` smart-constructor` should be defined in the `openai` module, rather than here.

Are we directly relying on this in Core after all? I thought the idea was to split this off. We should be relying on AIClient`, so this `openAI` function should be defined in the `openai` module

Reply:
That would be ideal, but based on our conversation on Slack yesterday I thought we still need the openAI deps in core if we want to provide a single module with sane defaults, and those defaults are the OpenAI ones.

Alternatively, you would have to bring a `bom` or two modules, xef core for the base, and the OpenAI runtime. Things like `getOrElse` may have to live in the OpenAI module because that operation decides what runtime it uses.

I think the AIClient over the scope is a bad abstraction because the client is dependent on each operation and may change with each operation depending on the model you use not the scope. I plan on completely getting read of the `AIClient` and trying to move those specific operations like `createChatCompletions` or `createImage` to the actual model hierarchy so chat and image models independently implement those.

Still, if we want xef-core to run independently without additional deps we need the OpenAI dep here or a different strategy.

Reply:
> That would be ideal, but based on our conversation on Slack yesterday I thought we still need the openAI deps in core if we want to provide a single module with sane defaults, and those defaults are the OpenAI ones.

I think there was some miscommunication there, `core` should expose some `interface` and `openAI` needs to expos some smart-constructor/implementation. Otherwise there is no point in splitting everything in different modules, if we end up enforcing depending on everything.

You wouldn't need to bring in two dependencies, since `OpenAI` would transitively bring in `core`, and unless versions start differing there is no need for `bom` IMO.

`getOrElse` is just there for backwards compatibility as requested by the other teams relying on Xef, it's nothing more than `try/catch` now so it can be removed entirely.

> Still, if we want xef-core to run independently without additional deps we need the OpenAI dep here or a different strategy.

Those decision are unrelated to technical side. Currently there is no clear vision of how we want this library to be used, or what the API should look like. We should answer that question first, and make the decisions in regards to that. If we want everyone to rely on `OpenAI` by default, or as a fallback, then it needs to be in `core`.